src/main/java/uk/gov/companieshouse/mapper/CompanyOfficerMapper.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.mapper;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.model.db.dissolution.DissolutionDirector;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CompanyOfficerMapper {
+    public boolean mapCompanyOfficersToIsOverHalfSelected(
+            List<CompanyOfficerApi> companyOfficers,
+            List<DissolutionDirector> selectedDirectors
+            ) {
+
+        final List<String> activeOfficers = this.mapCompanyOfficersToActiveList(companyOfficers);
+        final List<String> selectedOfficers = this.mapDissolutionDirectorsToSelectedOfficers(selectedDirectors);
+
+        final float officersSelected = selectedOfficers.stream().filter(
+                dissolutionDirector -> activeOfficers
+                        .stream()
+                        .anyMatch(companyOfficerApi -> companyOfficerApi.equals(dissolutionDirector)))
+                .count();
+        if (officersSelected == 0 || activeOfficers.isEmpty()) {
```

Comment:
you should be checking `activeOfficers.isEmpty()` before calculating `officersSelcted`